worker/src/request.rs

```diff
@@ -228,3 +229,28 @@ impl Request {
         &self.edge_request
     }
 }
+
+/// Used to add additional helper functions to url::Url
+pub trait UrlExt {
+    /// Given a query parameter, returns an Iterator of values for that parameter in the url's
+    /// query string
+    fn param<'a>(&'a self, key: &'a str) -> Box<dyn Iterator<Item = Cow<'a, str>> + 'a>;
```

Comment:
I think if we're going just for a QOL improvement it would make more sense for this to return the first value of that name. Wanting to get all values for a query param seems more niche to me and I'd rather make the more widely easier. I think we should change the method to:
```rust
fn param<'a>(&'a self, key: &'a str) -> Cow<'a, str>;
```

Reply:
I do have a use case for multiple parameters. Maybe having both `param` and `param_iter` would be fine? Though tbf, I could just use `Url::query_params`.